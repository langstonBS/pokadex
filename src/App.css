.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

ul {
  list-style-type: none;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}
.search {
  display: flex;
  justify-content: center;
  padding: 10px;
}

button{
  background-color: blueviolet;
  border: none;
  color: white;
  padding: 5px 30px;
  text-align: center;
  display: inline-block;
  font-size: 16px;
  margin: 15px;
  border-radius: 5px;
  transition-duration: 0.4s;
}
button:hover{
  background-color: rgb(70, 13, 122);
}
.searchList {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  background-color: blueviolet;
  padding-bottom: 20px;
  border-radius: 10px 10px 0 0;
  
}

select {
  position: relative;
  display: flex;
  width: 10em;
  height: 2em;
  line-height: 2;
  border-radius: .25em;
  transition-duration: 0.4s;
}

