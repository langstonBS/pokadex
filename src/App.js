import React, { Component } from 'react';
//import request from 'superagent';
import RenderPokemon from './RenderPokemon';

export default class App extends Component {

  state = {
    searchQuery: null,
    searchLevle: 0,
    getData: null,
    attacDefenc: 'attack',
  }
  handleChange = (e) => {
    const value = e.target.value;
    this.setState({ searchQuery: value });
    console.log(this.state)
  }
  levelChange = (e) => {
    let value = e.target.value;
    if (value <= 0|| value - (Math.floor(value)) !== 0) {
      value = 0;
    }
    this.setState({ searchLevle: value });
    console.log(this.state)
  }
  sortBy = (e) => {
    const value = e.target.value;
    this.setState({ attacDefenc: value })
    console.log(value);

  }

  handleClick = () => {
    const getData = 'I am clicked';
    console.log(getData);
}

  render() {
    return (
      <div>
        i am here
        <input onChange={this.handleChange} />
        <input type="number" pattern="[0-9]" onChange={this.levelChange} />
        <select className="selctionStyle" onChange={this.sortBy}>
          <option value="attack">attack</option>
          <option value="defend">defend</option>
        </select>
        < button onClick={this.handleClick}>serch</button>
        < RenderPokemon />
      </div>
    )
  }
}
