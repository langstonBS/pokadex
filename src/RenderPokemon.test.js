import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import Adapter from 'enzyme-adapter-react-16'



it('renders welcome message', () => {
  const wrapper = shallow(<App />);
  const welcome = <p>Edit <code>src/App.js</code> and save to reload.</p>;
  const containsWelcome = wrapper.contains(welcome)
  expect(containsWelcome).toEqual(true);
});