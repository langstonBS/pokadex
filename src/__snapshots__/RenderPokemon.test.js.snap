// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates profile 1`] = `ShallowWrapper {}`;
